scripts/get-test-pass-rate.js

```diff
@@ -0,0 +1,56 @@
+// This script is used in the "runAfterAll" stage in our E2E test pipeline. It's used 
+// to get timeline and metrics data so that the test pass rate can be calculated in a later step.
+
+const BUILD_ID = process.env.BUILD_ID;
+const ADO_API_TOKEN = process.env.ADO_API_TOKEN;
+const TEST_WORKSPACE = process.env.TEST_WORKSPACE;
+const BUILD_SOURCES_DIRECTORY = process.env.BUILD_SOURCES_DIRECTORY;
+console.log(BUILD_ID);
+console.log(TEST_WORKSPACE);
+// Create output folder - Note: This requires Node.js fs module
+const fs = require('fs');
+console.log("Starting fetch")
+// Fetch data from Timeline API
+const apiUrl = `https://dev.azure.com/fluidframework/internal/_apis/build/builds/${BUILD_ID}/timeline?api-version=7.1-preview.2`;
+if (!fs.existsSync(`${TEST_WORKSPACE}/stageFiles`)) {
+    fs.mkdirSync(`${TEST_WORKSPACE}/stageFiles`, { recursive: true });
+    console.log("Folder created");
+}
+
+let stages = [];
+fetch(apiUrl, {
+    headers: {
+        'Authorization': `Basic ${ADO_API_TOKEN}`
+    }
+})
+.then(response => {
```

Comment:
Let's make the file a `.mjs` file instead of `.js` to indicate it's a `module`, so we can use `const response = await fetch(...)` and we can avoid the Promise continuations, it should make the script easier to read and reason about. And it makes @Josmithr 's comment about throwing/exiting in the catch unnecessary, since an error will immediately terminate the script.